### PR TITLE
fuse: implement the FUSE_LOOKUP command

### PR DESCRIPTION
**Pull request by boyuan-he**
Implement inode.Lookup() for FUSE. 

Original PR [fs/fuse: implement inode.Lookup()](https://github.com/google/gvisor/pull/3261) by @avagin . Move since have many overlap code with previous sub marshallable and upcoming FUSE_MKNOD code.

### CODE REVIEW

File: pkg/sentry/fsimpl/fuse/fusefs.go [hunk 1]
```diff
@@ -227,6 +241,53 @@ func (i *inode) Open(ctx context.Context, rp *vfs.ResolvingPath, vfsd *vfs.Dentr
...
+func (inode) IterDirents(ctx context.Context, callback vfs.IterDirentsCallback, offset, relOffset int64) (int64, error) {
+	return offset, nil
+}
+
+// Valid implements Inode.Valid.
+func (inode) Valid(ctx context.Context) bool {
+	return true
+}
+
+// newEntry call FUSE server for entry creation and allocate corresponding entry according to response.
```
Comment (craig08) [inline pkg/sentry/fsimpl/fuse/fusefs.go:268]: calls, allocates

File: pkg/sentry/fsimpl/fuse/fusefs.go [hunk 2]
```diff
@@ -204,14 +204,28 @@ type inode struct {
...
+	i := &inode{fs: fs}
+	i.InodeAttrs.Init(creds, linux.UNNAMED_MAJOR, fs.devMinor, 1, linux.ModeDirectory|0755)
+	i.OrderedChildren.Init(kernfs.OrderedChildrenOptions{})
+	i.dentry.Init(i)
+	i.NodeID = 1
+
+	return &i.dentry
+}
+
+func (fs *filesystem) newInode(nodeID uint64, generation uint64, attr linux.FUSEAttr) *kernfs.Dentry {
```
Comment (craig08) [inline pkg/sentry/fsimpl/fuse/fusefs.go:225]: `generation` is not used in the function, do we have to consider it now?

Comment (boyuan-he) [reply pkg/sentry/fsimpl/fuse/fusefs.go:225]: Removed, probably can consider it when we actually use it

**Comment by avagin** [comment]
@boyuan-he please rebase this pr

File: pkg/abi/linux/fuse.go [hunk 3]
```diff
@@ -301,3 +306,54 @@ type FUSEGetAttrOut struct {
...
+type FUSELookupIn struct {
+	marshal.StubMarshallable
+
+	// Name is a file name to be looked up.
+	Name string
+}
+
+// MarshalUnsafe serializes r.name to the dst buffer.
+func (r *FUSELookupIn) MarshalUnsafe(buf []byte) {
+	copy(buf, []byte(r.Name))
```
Comment (jinmouil) [inline pkg/abi/linux/fuse.go:352]: Can we avoid making this copy of data?

Comment (boyuan-he) [reply pkg/abi/linux/fuse.go:352]: discussed offline

**Comment by mrahatm** [comment]
Merged as part of feature/fuse.

**Pull request closed without merging**